bears/js/ESLintBear.py

```diff
@@ -58,15 +58,28 @@ def generate_config(filename, file):
         return '{"extends": "eslint:recommended"}'
 
     def process_output(self, output, filename, file):
-        if output[1]:
+        # Taking output from stderr for ESLint 2 program errors.
+        # ESLint 2 is no longer supported, but it is here so that anyone
+        # with ESLint 2 will still see any errors that were emitted.
+        if output[1]:  # pragma: no cover
+            output = output[1]
```

Comment:
I think you can put the `# pragma` here, making it much more specific.

However, the previous code allowed for errors on stderr, which were reported with `self.warn`, *and* real results in json on stdout which were reported as `Result` objects.
Your modified approach only allows stderr _or_ stdout to be processed.

Reply:
I think if there's an error in stderr there shouldn't be any result?

But I'll modify it anyway just for precaution.

Reply:
From before:

> But, unfortunately we need ESLint 2 just to test this block

Yeah but I think because of the ESLint 3 requirement we don't need this block at all :O

Reply:
> Yeah but I think because of the ESLint 3 requirement we don't need this block at all

Everyone currently *is* using ESLint 2.x, with explicit pins.  either they are doing `npm install -g eslint=2.x`, or unlikely they are fetching and using the `package.json` out of the coala-bears repo , which isnt an easy option as it also requires a custom `NODE_PATH` (a problem which stumped us for months).

If someone upgrades their CI env to using coala 0.12 (i.e. with this patch in it), and doesnt update the eslint version in their CI, any eslint 2 errors would be silently discarded if stderr is disregarded.

This is why it is so important that coala get non-Python dependency management sorted out properly, so that it bombs out at the beginning of CI with a version error.

Reply:
Ahh ugh okay... :+1: